nemo/core/optim/lr_scheduler.py

```diff
@@ -877,6 +877,11 @@ def prepare_lr_scheduler(
                 batch_size = train_dataloader.batch_sampler.micro_batch_size
             else:
                 raise ValueError(f'Could not find batch_size from batch_sampler: {train_dataloader.batch_sampler}')
+        elif hasattr(train_dataloader, 'sampler') and train_dataloader.sampler is not None:
+            if train_dataloader.sampler.micro_batch_size is not None:
```

Comment:
Not guaranteed to have micro batch size in sampler, please do a hasattr check first

Reply:
I see the code above already does this part, so I don't see the need to have this new else case

Reply:
You should do a hasattr check for the existence of the micro batch variable before calling it